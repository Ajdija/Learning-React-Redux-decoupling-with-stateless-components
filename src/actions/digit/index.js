export const DIGIT_CLICK_ACTION = 'DIGIT_CLICK_ACTION';
export const CLICKED_DIGIT_KEY = 'CLICKED_DIGIT_KEY';

export const digitClick = digit => ({
  type: DIGIT_CLICK_ACTION,
  payload: { [CLICKED_DIGIT_KEY]: digit }
})
