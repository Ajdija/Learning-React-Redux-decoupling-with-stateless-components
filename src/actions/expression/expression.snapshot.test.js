import {
  expressionReducer as reducer,
  updateExpression,
  EXPRESSION_UPDATE_ACTION
} from './index';
import { digitClick, DIGIT_CLICK_ACTION } from '../digit';

describe('expression reducer', () => {
  it('should return initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it(`should handle ${EXPRESSION_UPDATE_ACTION} action`, () => {
    const action = updateExpression(10);
    const initialState = undefined;

    // First invoke
    const state1 = reducer(initialState, action);
    expect(state1).toMatchSnapshot();

    // Second invoke
    const action2 = updateExpression(15);
    const state2 = reducer(state1, action2);
    expect(state2).toMatchSnapshot();
  });

  it(`should handle ${DIGIT_CLICK_ACTION} action`, () => {
    const action = digitClick(0);
    const initialState = undefined;

    // First invoke
    const state1 = reducer(initialState, action);
    expect(state1).toMatchSnapshot();

    // Second invoke
    const action2 = digitClick(5);
    const state2 = reducer(state1, action2);
    expect(state2).toMatchSnapshot();
  });
});
