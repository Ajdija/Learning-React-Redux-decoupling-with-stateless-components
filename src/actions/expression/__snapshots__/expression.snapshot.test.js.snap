// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expression reducer should handle DIGIT_CLICK_ACTION action 1`] = `0`;

exports[`expression reducer should handle DIGIT_CLICK_ACTION action 2`] = `5`;

exports[`expression reducer should handle EXPRESSION_UPDATE_ACTION action 1`] = `10`;

exports[`expression reducer should handle EXPRESSION_UPDATE_ACTION action 2`] = `15`;

exports[`expression reducer should return initial state 1`] = `0`;
